clustering/infinispan/extension/src/main/java/org/jboss/as/clustering/infinispan/subsystem/CacheContainerBuilder.java

```diff
@@ -109,7 +113,7 @@ public CacheContainerBuilder configure(OperationContext context, ModelNode model
         };
         Service<CacheContainer> service = new SuppliedValueService<>(mapper, supplier, destroyer);
         ServiceBuilder<CacheContainer> builder = target.addService(this.getServiceName(), service)
-                .addAliases(this.aliases.stream().toArray(ServiceName[]::new))
+                .addAliases(aliases.toArray(new ServiceName[0]))
```

Comment:
Why create an unnecessary new object with the wrong size?  We know the size, just allocate it appropriately.

Reply:
This is actually faster because it skips pre-zeroing of the array, and the new ServiceName[0] might even be deleted.  But for extra safety, hoisting it up to a constant is probably a good idea.

See https://shipilev.net/blog/2016/arrays-wisdom-ancients for more info.

Reply:
Interesting... Thanks for the link!